x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -34,182 +34,281 @@ func EvidenceSignBytes(sequence uint64, data []byte) []byte {
 }
 
 // HeaderSignBytes returns the sign bytes for verification of misbehaviour.
-//
-// Format: {sequence}{header.newPubKey}
-func HeaderSignBytes(header *Header) []byte {
-	return append(
-		sdk.Uint64ToBigEndian(header.Sequence),
-		header.GetPubKey().Bytes()...,
-	)
+func HeaderSignBytes(
+	cdc codec.BinaryMarshaler,
+	header *Header,
+) ([]byte, error) {
+	data := &HeaderData{
+		NewPubKey:      header.NewPublicKey,
+		NewDiversifier: header.NewDiversifier,
+	}
+
+	dataBz, err := cdc.MarshalBinaryBare(data)
+	if err != nil {
+		return nil, err
+	}
+
+	signBytes := &SignBytes{
+		Sequence:    header.Sequence,
+		Timestamp:   header.Timestamp,
+		Diversifier: header.NewDiversifier, // TODO: double check if it needs to be passed from current consensus state
```

Comment:
I think this is correct. We just want to ensure the old public key signed over the new public key and new diversifier

Reply:
we should remove the TODO